src/plugins/region_map/server/services/opensearch_service.ts

```diff
@@ -3,8 +3,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-export default class OpensearchService {
-  constructor(esDriver) {
+import { ILegacyCustomClusterClient } from 'opensearch-dashboards/server';
+
+// eslint-disable-next-line import/no-default-export
+export default class OpenSearchService {
+  constructor(esDriver: ILegacyCustomClusterClient) {
     this.esDriver = esDriver;
```

Comment:
`esDriver` doesn't exist on OpenSearchService from the compiler's perspective but once it's compiled down to JS it won't care. But we have to goal to not make things worse with types in our repo can we make sure to include types for parameters in the file and ensure within this context variables are defined before used.

Reply:
Updated, got it. I ran ` tsc --noEmit && eslint` and found out all other ts issues too!